### PR TITLE
Clazy: fix-qlatin1string-allocations

### PR DESCRIPTION
**Pull request by TheOneRing**

### CODE REVIEW

File: src/gui/shareusergroupwidget.cpp [hunk 1]
```diff
@@ -464,7 +465,7 @@ void ShareUserLine::slotAvatarLoaded(const QPixmap &avatar)
     _ui->avatar->setPixmap(_avatar);
 
     // Remove the stylesheet for the fallback avatar
-    _ui->avatar->setStyleSheet("");
+    _ui->avatar->setStyleSheet(QLatin1String(""));
```
Comment (TheOneRing) [inline src/gui/shareusergroupwidget.cpp:468]:
```suggestion
    _ui->avatar->setStyleSheet(QString());
```

File: src/gui/socketapi/socketapi.cpp [hunk 2]
```diff
@@ -453,7 +453,7 @@ void SocketApi::slotUpdateFolderView(Folder *f)
             Q_FALLTHROUGH();
         case SyncResult::Error: {
             const QString rootPath = Utility::stripTrailingSlash(f->path());
-            broadcastStatusPushMessage(rootPath, f->syncEngine().syncFileStatusTracker().fileStatus(""));
+            broadcastStatusPushMessage(rootPath, f->syncEngine().syncFileStatusTracker().fileStatus(QLatin1String("")));
```
Comment (TheOneRing) [inline src/gui/socketapi/socketapi.cpp:456]:
```suggestion
            broadcastStatusPushMessage(rootPath, f->syncEngine().syncFileStatusTracker().fileStatus(QString()));
```

File: src/gui/ocssharejob.h [hunk 3]
```diff
@@ -106,7 +106,7 @@ namespace OcsShareJob {
      */
     JsonApiJob *createShare(AccountPtr account, QObject *parent, const QString &path,
         const Share::ShareType shareType,
-        const QString &shareWith = "",
+        const QString &shareWith = QLatin1String(""),
```
Comment (TheOneRing) [inline src/gui/ocssharejob.h:109]:
```suggestion
        const QString &shareWith = QString(),
```

File: src/gui/shareusergroupwidget.cpp [hunk 4]
```diff
@@ -164,7 +164,7 @@ void ShareUserGroupWidget::searchForSharees()
     ShareeModel::ShareeSet blacklist;
 
     // Add the current user to _sharees since we can't share with ourself
-    QSharedPointer<Sharee> currentUser(new Sharee(_account->credentials()->user(), "", Sharee::Type::User));
+    QSharedPointer<Sharee> currentUser(new Sharee(_account->credentials()->user(), QLatin1String(""), Sharee::Type::User));
```
Comment (TheOneRing) [inline src/gui/shareusergroupwidget.cpp:167]:
```suggestion
    QSharedPointer<Sharee> currentUser(new Sharee(_account->credentials()->user(), QString(), Sharee::Type::User));
```

File: src/gui/shareusergroupwidget.cpp [hunk 5]
```diff
@@ -426,14 +426,15 @@ void ShareUserLine::loadAvatar()
...
+    const QString style = QStringLiteral(R"(* {
         color: #fff;
         background-color: %1;
         border-radius: %2px;
         text-align: center;
         line-height: %2px;
         font-size: %2px;
-    })").arg(bg.name(), QString::number(avatarSize / 2));
+    })")
+                              .arg(bg.name(), QString::number(avatarSize / 2));
```
Comment (fmoc) [inline src/gui/shareusergroupwidget.cpp:437]: Formatting is off.

**Review by fmoc — CHANGES_REQUESTED** [review]

**Pull request merged**